libs/sdk-ui-dashboard/src/presentation/scheduledEmail/DefaultScheduledEmailManagementDialog/useScheduledEmailManagement.ts

```diff
@@ -1,48 +1,58 @@
 // (C) 2022 GoodData Corporation
 
+import { IScheduledMail } from "@gooddata/sdk-backend-spi";
 import {
     GoodDataSdkError,
     useBackendStrict,
     useCancelablePromise,
     useWorkspaceStrict,
 } from "@gooddata/sdk-ui";
-import {
-    selectCanManageScheduledMail,
-    selectCurrentUser,
-    selectDashboardRef,
-    useDashboardSelector,
-} from "../../../model";
+import { selectCanManageScheduledMail, selectDashboardRef, useDashboardSelector } from "../../../model";
 
 interface IUseScheduledEmailManagementProps {
     /**
      * Callback to be called, when emails fail to load.
      */
     onError?: (error: GoodDataSdkError) => void;
+
+    /**
+     * Callback to be called, when emails load.
+     */
+    onSuccess: (scheduledEmails: IScheduledMail[]) => void;
+
+    /**
+     * Flag to handle data refetching logic
+     */
+    needsReload: boolean;
 }
 
 export const useScheduledEmailManagement = (props: IUseScheduledEmailManagementProps) => {
+    const { onError, onSuccess, needsReload } = props;
     const dashboardRef = useDashboardSelector(selectDashboardRef);
-    const currentUser = useDashboardSelector(selectCurrentUser);
     const canManageScheduledMail = useDashboardSelector(selectCanManageScheduledMail);
     const effectiveBackend = useBackendStrict();
     const effectiveWorkspace = useWorkspaceStrict();
 
-    const promise = async () => {
-        const scheduledEmails = await effectiveBackend
-            .workspace(effectiveWorkspace)
-            .dashboards()
-            .getScheduledMailsForDashboard(dashboardRef!, {
-                createdByCurrentUser: !canManageScheduledMail,
-            });
+    const promise = needsReload
+        ? async () => {
+              const scheduledEmails = await effectiveBackend
```

Comment:
don't we need some memoization to make sure it's not loaded multiple times when it's called multiple times with needsReload `true`

Reply:
This is already handled by setting the dependency array of `useCancelablePromise` hook